includes/installer/class-installer.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * File containing the class \Sensei\Installer\Installer.
+ *
+ * @package sensei
+ * @since $$next-version$$
+ */
+
+namespace Sensei\Installer;
+
+use Sensei_Updates;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Installer class.
+ *
+ * Responsible for running DB updates that need to be run per version.
+ *
+ * @since $$next-version$$
+ */
+class Installer {
+
+	private const SENSEI_VERSION_OPTION_NAME = 'sensei-version';
+
+	/**
+	 * Instance of the class.
+	 *
+	 * @since $$next-version$$
+	 * @var self
+	 */
+	private static $instance;
+
+	/**
+	 * The database schema class.
+	 *
+	 * @since $$next-version$$
+	 * @var Schema
+	 */
+	private $schema;
+
+	/**
+	 * Current Sensei version.
+	 *
+	 * @var string|null
+	 */
+	private $version;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param Schema      $schema Schema migration object.
+	 * @param string|null $version Current Sensei version.
+	 */
+	public function __construct( Schema $schema, ?string $version ) {
+		$this->schema  = $schema;
+		$this->version = $version;
+	}
+
+	/**
+	 * Fetches an instance of the class.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string|null $version Current Sensei version.
+	 * @return self
+	 */
+	public static function instance( ?string $version = null ): self {
+		if ( ! self::$instance ) {
+			self::$instance = new self( new Schema(), $version );
+		}
+
+		return self::$instance;
+	}
+
+	/**
+	 * Initialize necessary hooks.
+	 *
+	 * @since $$next-version$$
+	 */
+	public function init() {
+		register_activation_hook( SENSEI_LMS_PLUGIN_FILE, [ $this, 'install' ] );
+		add_action( 'plugins_loaded', [ $this, 'install' ] );
+		add_action( 'init', array( $this, 'update' ) );
```

Comment:
I think it might be important to have a test that checks if these hooks are triggering.